numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
```

Comment:
the concern is for example we have:

```
#ifdef GCC_VERSION > 4.2
#define npy_copysign copysign
#endif
```

now a compiler comes along and pretends to be gcc 4.2 but doesn't have the copysign builtin and the system doesn't have one either you get a compile error/undefined reference.
As compiler version checks are the only way to determine if a builtin exists with gcc this will always be fragile. (Of course one could change the check to only inline with compilers that offer a good way to detect this)

In practice this is probably not such a big deal but there just also wasn't a good reason to bother either. Except for some of the floating point property functions all math functions are so slow that inlining didn't make much difference
Now that vector math libraries are more common it might be worth the unlikely case of odd configurations breaking.